code/datums/status_effects/buffs.dm

```diff
@@ -472,3 +472,62 @@
 	else
 		to_chat(owner, "<span class='cultitalic'>[pick(un_hopeful_messages)]</span>")
 
+/datum/status_effect/thrall_net
+	id = "thrall_net"
+	tick_interval = 2 SECONDS
+	duration = -1
+	alert_type = null
+	var/blood_cost_per_tick = 5
+	var/list/target_UIDs = list()
+	var/datum/antagonist/vampire/vamp
+
+/datum/status_effect/thrall_net/on_creation(mob/living/new_owner, datum/antagonist/vampire/V, ...)
+	vamp = V
+	. = ..()
+	START_PROCESSING(SSfastprocess, src)
+	target_UIDs += owner.UID()
+	for(var/datum/mind/M in owner.mind.som.serv)
+		if(!M.has_antag_datum(/datum/antagonist/mindslave/thrall))
+			continue
+		if(!(M.current in view(7, owner)))
+			continue
+		target_UIDs += M.current.UID()
+		M.current.Beam(owner, "sendbeam", time = 2 SECONDS, maxdistance = 7)
+
+/datum/status_effect/thrall_net/tick()
+	var/total_damage = 0
+	for(var/thing in target_UIDs)
+		var/mob/living/L = locateUID(thing)
+		if(!(L in view(7, owner)))
+			target_UIDs -= thing
+			continue
+		total_damage += (L.maxHealth - L.health)
```

Comment:
Not sure if this is intended but I'd assume not based on the PR description. You will actually heal damage initially from using the ability as it is based on your max health rather than current health, as it distributed. Maybe you did it this way due to code complexity? This could be pretty huge at healing a lot of damage given the right circumstances.